core/src/main/java/org/elasticsearch/common/unit/Fuzziness.java

```diff
@@ -187,19 +193,24 @@ public byte asByte() {
     }
 
     public double asDouble() {
-        if (this == AUTO) {
+        if (isAuto()) {
             return 1d;
         }
         return Double.parseDouble(fuzziness.toString());
     }
 
     public float asFloat() {
-        if (this == AUTO) {
+        if (isAuto()) {
             return 1f;
         }
         return Float.parseFloat(fuzziness.toString());
     }
 
+    // A deserialized fuzziness with AUTO is not the static Fuzziness.AUTO object
+    private boolean isAuto() {
+        return this == AUTO || fuzziness.equalsIgnoreCase("AUTO");
```

Comment:
then I guess the equals method is wrong too cause auto != AUTO . I think I would try to uppercase (or lowercase) when parsing and make this case insensitive that way.